### DISCUSSION TITLE
Problems updating Tine2.0

### DISCUSSION THREAD

**Discussion by selfarian** [tine20/tine20#7336] [Q&A] [answered]
Hello,

i had problems with tine, after downloading the newest Version (i had php 7.3). Now i updated to php 7.4 and tried to run setup php and all checks are ok, but when I try to update the applications, i get an "Setup Admin undefined" error:
An exception occurred while executing 'ALTER TABLE tine20_tree_nodes ADD linkto VARCHAR(40) DEFAULT NULL, CHANGE name name VARCHAR(255) NOT NULL COLLATE `utf8mb4_bin`': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes

.../vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(182): Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException()
.../vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(159): Doctrine\DBAL\DBALException::wrapException()
.../vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(2225): Doctrine\DBAL\DBALException::driverExceptionDuringQuery()
.../vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(1312): Doctrine\DBAL\Connection->handleExceptionDuringQuery()
.../vendor/doctrine/orm/lib/Doctrine/ORM/Tools/SchemaTool.php(922): Doctrine\DBAL\Connection->executeQuery()
.../Setup/SchemaTool.php(146): Doctrine\ORM\Tools\SchemaTool->updateSchema()
.../Tinebase/Setup/Update/14.php(93): Setup_SchemaTool::updateSchema()
.../Setup/Controller.php(556): Tinebase_Setup_Update_14->update001()
.../Setup/Frontend/Json.php(118): Setup_Controller->updateApplications()
.../vendor/zendframework/zendframework1/library/Zend/Server/Abstract.php(231): Setup_Frontend_Json->updateApplications()
.../vendor/zendframework/zendframework1/library/Zend/Json/Server.php(603): Zend_Server_Abstract->_dispatch()
.../vendor/zendframework/zendframework1/library/Zend/Json/Server.php(206): Zend_Json_Server->_handle()
.../Setup/Server/Json.php(90): Zend_Json_Server->handle()
.../Setup/Core.php(117): Setup_Server_Json->handle()
.../setup.php(14): Setup_Core::dispatchRequest()

Has someone an idea how to fix this?

Thank you very much!

**Comment by pschuele** [comment, marked as answer]
hi,
it seems that your mysql (mariadb?) version is too low. can you check your version and update it if lower than the requirements ([MySQL]5.7 - 8.0 or [MariaDB] 10.2 - 10.6)?
and: sorry for the late reply :blush: 
Philipp

**Comment by selfarian** [comment]
Hi Philipp,

no problem. Thanks for your advice. I think you are right. I have mysql 5.5, so its very old. Problem is, i need to do a distribution upgrade to use newer versions :/ so i have to fix my issues with this and then mysql and then tine. Hopefully this will do it :)

Thank you!
Am 18.05.2022 11:35:14, schrieb Philipp Schüle:

> 

> hi,
> 

it seems that your mysql (mariadb?) version is too low. can you check your version and update it if lower than the requirements ([MySQL]5.7 - 8.0 or [MariaDB] 10.2 - 10.6)?
> 

and: sorry for the late reply ?
> 

Philipp>
